src/build_tools.rs

```diff
@@ -133,13 +139,25 @@ impl SchemaError {
     }
 
     pub fn from_val_error(py: Python, error: ValError) -> PyErr {
-        match error {
-            ValError::LineErrors(line_errors) => {
-                let details = pretty_line_errors(py, line_errors);
-                SchemaError::new_err(format!("Invalid Schema:\n{details}"))
+        let validation_error = ValidationError::from_val_error(py, "Schema".to_object(py), error, None);
+        match validation_error.into_value(py).extract::<ValidationError>(py) {
+            Ok(err) => {
+                let schema_error = SchemaError {
+                    value: SchemaErrorInner::Error(err),
+                };
+                match Py::new(py, schema_error) {
+                    Ok(err) => PyErr::from_value(err.into_ref(py)),
+                    Err(err) => err,
+                }
             }
-            ValError::InternalErr(py_err) => py_err,
-            ValError::Omit => unreachable!(),
+            Err(err) => err,
+        }
+    }
+
+    fn message(&self) -> String {
+        match &self.value {
+            SchemaErrorInner::Message(message) => message.to_owned(),
+            SchemaErrorInner::Error(_) => "<ValidationError>".to_owned(),
         }
```

Comment:
This `fn message` is the main thing that sticks out to me as obviously weird; happy to implement something a bit different with some guidance. (Maybe drop this and just implement the Display and Debug directly, as this is the only place it is used, but not exactly sure what to use instead — `format!("{}", error)`/`format!("{:?}", error)` for rendering the `ValidationError` instead of `"<ValidationError>"`?

Reply:
i think it should be the output of `display()`, see my comment below.

Reply:
on reflection, I think this is fine.